CRM/Contribute/BAO/Contribution.php

```diff
@@ -436,6 +420,22 @@ public static function create(&$params, $ids = array()) {
 
     $transaction->commit();
 
```

Comment:
I guess it should return if it rolls back - not continue - but then - we actually want it to fail really hard in this case - because basically something unexpected & probably data losing has just happened. So, safe might actually be a fatal.